### ISSUE TITLE
Windows Builds

### ISSUE THREAD

**Issue by Starz0r** [Peltoche/lsd#188]
Can we get some compiled builds for this application on the GitHub Releases page? I personally wouldn't mind compiling it for x86-64 and i686 for version 0.13.0 and 0.14.0 if it means it would land on the Releases page. Currently only Apple Macintosh and GNU/Linux are distributed, I feel it would be a lot more convenient for Windows Users if pre-compiled builds in some form were available to us.

**Comment by Peltoche** [comment]
Hi @Starz0r,

I totally agree with you. This is what I try to do with #146 but for now it's not successful. If you have some experiences with the rust compilation on travis some feedbacks would be welcomed.

**Comment by Starz0r** [comment]
I've worked with Travis before, but I've never done it with Rust. What I was talking about specifically was just compiling on a personal computer for immediate use with the current two release where Windows is actually supported. I can look into Travis with Rust after adding builds to the releases page, but I think this would be a better job for AppVeyor when it comes to Windows anyway.

**Comment by thorstenkampe** [comment]
Looking forward to a Windows binary, too...

**Comment by plaptov** [comment]
Any updates?

**Comment by Peltoche** [comment]
Nop, I didn't have the time to work on it. There is an appveyor pipeline running (https://ci.appveyor.com/project/Peltoche/lsd) but I didn't take the time to make it work. As it doesn't work I can't make it mandatory before merging so now it had accumulated some errors.

It would be nice if someone would take the point seriously. Maybe I will work on it but no promises, I have much less time to put in this project now. If someone need something in order to fix it, don't hesitate.

**Comment by Starz0r** [comment]
I can personally compile it on my machine for all supported versions without AppVeyor for now. It's really just about having builds that are available for package managers on Windows.

**Comment by rivy** [comment]
@Peltoche , I created a github workflow that can create Windows builds.

I iterated on it for the last couple of weeks and generated a workflow that mimics your complete build process. It contains style tests, MinSRV, as well as building and saving artifacts for all the prior platforms and, now, Windows.

Being a completionist, I also added code coverage (optionally uploaded to CodeCov.io) and automatic deployment of packages (including `dpkg` packages) to Github Releases for any commit tagged with a release number.

I've just opened a [PR](https://github.com/Peltoche/lsd/pull/308) with the changes.
